tracer/src/Datadog.Trace/Span.cs

```diff
@@ -409,26 +419,15 @@ internal string GetTag(string key)
 
         internal void Finish(TimeSpan duration)
         {
-            var shouldCloseSpan = false;
-            lock (_lock)
+            _resourceName ??= OperationName;
```

Comment:
Why did you add a backing field to `ResourceName`? AFAIK, this doesn't make it any more thread-safe right? 🤔 Did you mean to use `Interlocked.CompareExchange`?

Reply:
Oh good question, I initially used the interlocked approach, but then I realise that this shouldn't be thread safe at all. It's an extreme edge case that I think it doesn't worth it.  I'll remove the backing field.

Reply:
done in [84ce8de](https://github.com/DataDog/dd-trace-dotnet/pull/4125/commits/84ce8de1223471d73f08d97452e7fdf9cd483075)